codec-http/src/test/java/io/netty/handler/codec/http/HttpServerUpgradeHandlerTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.junit.Test;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.http.HttpServerUpgradeHandler.UpgradeCodec;
+import io.netty.handler.codec.http.HttpServerUpgradeHandler.UpgradeCodecFactory;
+import io.netty.util.CharsetUtil;
+
+import static org.junit.Assert.*;
+
+public class HttpServerUpgradeHandlerTest {
```

Comment:
this test LGTM, but as a sanity check, does it fail without your change?

Reply:
No, not anymore. Originally it did, but now that the `write` override on line 96 calls `super.write` as @normanmaurer requested, it will succeed either way. See his second round of comments.

I'm happy to take suggestions on how to test this cleanly, but the only thing I can think of is to drop the promise (what I had originally, and it did fail without my patch) or to store it somewhere else and fail it later with the former being bad form and the latter being a pretty ugly mess.

Reply:
Using normans suggestion it now fails without the changes to `HttpServerUpgradeHandler`.

Reply:
@bryce-anderson did you forget to push this change ?

Reply:
@normanmaurer yes, I had. Sorry about that.